src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -48,19 +48,19 @@ partitions_request_iterator validate_replicas(
 
     if (valid_range_end != end) {
         vlog(klog.debug, "Invalid replicas: ec {} - {}", ec, error_message);
+        std::transform(
```

Comment:
if `valid_range_end == end` won't that just do nothing? Is the above conditional needed?

Reply:
great question. maybe it's to keep all the work together, and check was already  needed for the vlog line?

Reply:
It was two fold:
1. put the work and vlog line together
2. There was a problem where more data than expected was put into the response. The culprit was this call to std::transform, but now I cannot seem to remember where that was coming from. I'll retest and update in the next push.

Reply:
This change is unnecessary, I reverted it and retested. No failures.

Reply:
the flip side of this is that the change looked merely unnecessary. if when removed a failure went away, then there is likely still some other underlying issue.

Reply:
The failure was because of extra inserts into the response. I erroneously thought the problem was in this call to `std::transform`. Instead the problem was that I was always inserting `invalid_partitions` vector into the response. Commit https://github.com/redpanda-data/redpanda/pull/8414/commits/e0ddb9f56bd9d8238a1d5b9c08469f788eaa4e31 is the solution.

TLDR - There should not be an underlying issue here.